File: report/src/test/java/com/hartwig/actin/report/pdf/util/FormatsTest.java
package com.hartwig.actin.report.pdf.util;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class FormatsTest {

    @Test
    public void canFormatNumbers() {
        assertEquals("2.12", Formats.twoDigitNumber(2.123));
        assertEquals("2.12", Formats.twoDigitNumber(2.12));
        assertEquals("2.1", Formats.twoDigitNumber(2.1));
        assertEquals("2", Formats.twoDigitNumber(2.0));
    }
}

File: report/src/test/kotlin/com/hartwig/actin/report/pdf/util/FormatsTest.kt
package com.hartwig.actin.report.pdf.util

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class FormatsTest {

    @Test
    fun `Should correctly format numbers to two digits`() {
        assertThat(Formats.twoDigitNumber(2.123)).isEqualTo("2.12")
        assertThat(Formats.twoDigitNumber(2.12)).isEqualTo("2.12")
        assertThat(Formats.twoDigitNumber(2.1)).isEqualTo("2.1")
        assertThat(Formats.twoDigitNumber(2.0)).isEqualTo("2")
    }

    @Test
    fun `Should force a single digit when required`() {
        assertThat(Formats.forcedSingleDigitNumber(2.123)).isEqualTo("2.1")
        assertThat(Formats.forcedSingleDigitNumber(2.12)).isEqualTo("2.1")
        assertThat(Formats.forcedSingleDigitNumber(2.1)).isEqualTo("2.1")
        assertThat(Formats.forcedSingleDigitNumber(2.0)).isEqualTo("2.0")
        assertThat(Formats.forcedSingleDigitNumber(0.0)).isEqualTo("0.0")
    }
}